src/data/datacube/seriestype.cpp

```diff
@@ -25,11 +25,13 @@ const std::initializer_list<SeriesType> SeriesType::constTypes
 	SeriesType::Distinct
 };
 
-SeriesType SeriesType::fromString(const std::string &name, bool throws)
+SeriesType SeriesType::fromString(std::string_view name, bool throws)
 {
 	for (auto type : constTypes)
 		if (name == type.name) return type;
 
-	if (throws) throw std::logic_error("not recognized series type: " + name);
+	if (throws) throw std::logic_error(
+		"not recognized series type: " + std::string(name));
```

Comment:
probably `std::format` is preferable here.

Reply:
#include <format>
fatal error: 'format' file not found

Reply:
em++ -v
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 2.0.24 (416685fb964c14cde4be3e8a45ad26d75bac3e33)
clang version 13.0.0 (https://github.com/llvm/llvm-project 91f147792e815d401ae408989992f3c1530cc18a)

Reply:
OK. Thx, then no other changes required